lib/utils.js

```diff
@@ -95,14 +102,20 @@ const utils = {
         utils.logError('Configuration file missing for this directory, run losant configure to generate this file.');
         return;
       }
-      config.file = file;
+      config.file = foundFile;
     }
     return config;
   },
 
   loadConfig: async (file) => {
-    const userConfig = await utils.loadUserConfig(file);
-    const appConfig = await utils.loadApplicationConfig(file);
+    let appConfig = await utils.loadApplicationConfig(file);
+    const apiUrl = appConfig && appConfig.apiUrl || DEFAULT_URL;
+    const userConfig = await utils.loadUserConfig(apiUrl);
+    if (appConfig && (!appConfig.appUrl || !appConfig.endpointDomain)) {
+      const { endpointDomain, appUrl } = await utils.getWhitelabel({ apiToken: userConfig.apiToken });
```

Comment:
this whitelabel info could be stored in the user credential file - it is the same across every app for the same api url